sdk/deepbook-v3/src/transactions/deepbook.ts

```diff
@@ -0,0 +1,549 @@
+			arguments: [
+				tx.object(pool.address),
+				tx.object(balanceManager.address),
+				tradeProof,
+				tx.object(SUI_CLOCK_OBJECT_ID),
+			],
+			typeArguments: [baseCoin.type, quoteCoin.type],
+		});
+	};
+
+	withdrawSettledAmounts = (pool: Pool, balanceManager: BalanceManager) => (tx: Transaction) => {
+		const baseCoin = this.#config.getCoin(pool.baseCoin);
+		const quoteCoin = this.#config.getCoin(pool.quoteCoin);
+		const tradeProof = this.#config.balanceManager.generateProof(balanceManager);
+
+		tx.moveCall({
+			target: `${this.#config.DEEPBOOK_PACKAGE_ID}::pool::withdraw_settled_amounts`,
+			arguments: [tx.object(pool.address), tx.object(balanceManager.address), tradeProof],
+			typeArguments: [baseCoin.type, quoteCoin.type],
+		});
+	};
+
+	addDeepPricePoint = (targetPool: Pool, referencePool: Pool) => (tx: Transaction) => {
+		const targetBaseCoin = this.#config.getCoin(targetPool.baseCoin);
+		const targetQuoteCoin = this.#config.getCoin(targetPool.quoteCoin);
+		const referenceBaseCoin = this.#config.getCoin(referencePool.baseCoin);
+		const referenceQuoteCoin = this.#config.getCoin(referencePool.quoteCoin);
+		tx.moveCall({
+			target: `${this.#config.DEEPBOOK_PACKAGE_ID}::pool::add_deep_price_point`,
+			arguments: [
+				tx.object(targetPool.address),
+				tx.object(referencePool.address),
+				tx.object(SUI_CLOCK_OBJECT_ID),
+			],
+			typeArguments: [
+				targetBaseCoin.type,
+				targetQuoteCoin.type,
+				referenceBaseCoin.type,
+				referenceQuoteCoin.type,
+			],
+		});
+	};
+
+	claimRebates = (pool: Pool, balanceManager: BalanceManager) => (tx: Transaction) => {
+		const baseCoin = this.#config.getCoin(pool.baseCoin);
+		const quoteCoin = this.#config.getCoin(pool.quoteCoin);
+		const tradeProof = this.#config.balanceManager.generateProof(balanceManager);
+
+		tx.moveCall({
+			target: `${this.#config.DEEPBOOK_PACKAGE_ID}::pool::claim_rebates`,
+			arguments: [tx.object(pool.address), tx.object(balanceManager.address), tradeProof],
+			typeArguments: [baseCoin.type, quoteCoin.type],
+		});
+	};
+
+	burnDeep = (pool: Pool) => (tx: Transaction) => {
+		const baseCoin = this.#config.getCoin(pool.baseCoin);
+		const quoteCoin = this.#config.getCoin(pool.quoteCoin);
+		tx.moveCall({
+			target: `${this.#config.DEEPBOOK_PACKAGE_ID}::pool::burn_deep`,
+			arguments: [tx.object(pool.address), tx.object(this.#config.DEEP_TREASURY_ID)],
+			typeArguments: [baseCoin.type, quoteCoin.type],
+		});
+	};
+
+	midPrice = async (pool: Pool) => {
+		const tx = new Transaction();
+		const baseCoin = this.#config.getCoin(pool.baseCoin);
+		const quoteCoin = this.#config.getCoin(pool.quoteCoin);
+		const quoteScalar = quoteCoin.scalar;
+
+		tx.moveCall({
+			target: `${this.#config.DEEPBOOK_PACKAGE_ID}::pool::mid_price`,
+			arguments: [tx.object(pool.address), tx.object(SUI_CLOCK_OBJECT_ID)],
+			typeArguments: [baseCoin.type, quoteCoin.type],
+		});
+		const res = await this.#config.client.devInspectTransactionBlock({
```

Comment:
this methods that use devInspect probably don't belong in the contract (or the devInspect call should be hoisted up to the client level)